### PR TITLE
feat: enable revealed-value proofs

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
Enabled revealed-value proofs for all output types, which are be controlled via the consensus constants. Range-proof types are mapped to output types.

From the merge mining proxy/miner log, coinbase output showing `range_proof_type: RevealedValue` and `Proof: None`:

image

_**Note:** The faucets changed due to `RangeProofType` now being parsed with snake case - `"range_proof_type":"RevealedValue"` changed to `"range_proof_type":"revealed_value"` on every output line._

Motivation and Context
---
See #5968

How Has This Been Tested?
---
Added unit tests
Unit tests pass
Cucumber tests pass
System-level tests ~~[**TO BE COMPLETED**]~~

What process can a PR reviewer use to test or verify this change?
---
Code walkthrough
Review the additional unit tests
Run system-level tests

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 1]
```diff
@@ -672,8 +672,33 @@ impl ConsensusConstants {
...
+                RangeProofType::RevealedValue,
+            ]),
+            (OutputType::ValidatorNodeRegistration, &[
+                RangeProofType::BulletProofPlus,
+                RangeProofType::RevealedValue,
+            ]),
+            (OutputType::CodeTemplateRegistration, &[RangeProofType::BulletProofPlus]),
+        ]
+    }
+
```
Comment (hansieodendaal) [inline base_layer/core/src/consensus/consensus_constants.rs]: This must be reviewed and debated - this is an example. Easy to change the mapping that we want.

Comment (hansieodendaal) [reply base_layer/core/src/consensus/consensus_constants.rs]: General consensus: For now, we will only allow `OutputType::Coinbase` to have `RangeProofType::RevealedValue` as well; all other output types will only be `RangeProofType::BulletProofPlus`.

File: applications/minotari_miner/src/config.rs [hunk 2]
```diff
@@ -110,6 +113,7 @@ impl Default for MinerConfig {
             wait_timeout_on_error: 10,
             wallet_payment_address: TariAddress::default().to_hex(),
             stealth_payment: true,
+            range_proof_type: RangeProofType::BulletProofPlus,
```
Comment (SWvheerden) [inline applications/minotari_miner/src/config.rs]: can we make the default revealed.

**Review by SWvheerden — COMMENTED** [review]
This looks good, I think we just need to make the defaults revealed value so we can test them

File: applications/minotari_miner/src/config.rs (hunk 2 above)
Comment (hansieodendaal) [reply applications/minotari_miner/src/config.rs]: Let's just get @stringhandler's opinion as well

Comment (stringhandler) [reply applications/minotari_miner/src/config.rs]: Default revealed please.

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**